packages/playwright-core/src/server/android/android.ts

```diff
@@ -336,22 +335,35 @@ export class AndroidDevice extends SdkObject {
   }
 
   private async _refreshWebViews() {
+    // possible socketName, eg: webview_devtools_remote_32327, webview_devtools_remote_32327_zeus, webview_devtools_remote_zeus
     const sockets = (await this._backend.runCommand(`shell:cat /proc/net/unix | grep webview_devtools_remote`)).toString().split('\n');
     if (this._isClosed)
       return;
 
-    const newPids = new Set<number>();
+    const socketNames = new Set<string>();
     for (const line of sockets) {
-      const match = line.match(/[^@]+@webview_devtools_remote_(\d+)/);
-      if (!match)
+      const match = line.match(/[^@]+@.*?webview_devtools_remote_?(\d*)/);
+      const matchSocketName = line.match(/[^@]+@(.*?webview_devtools_remote_?.*)/);
+
+      if (!matchSocketName)
         continue;
-      const pid = +match[1];
-      newPids.add(pid);
-    }
-    for (const pid of newPids) {
-      if (this._webViews.has(pid))
+
+      const socketName = matchSocketName[1];
+      socketNames.add(socketName);
+
+      if (this._webViews.has(socketName))
         continue;
 
+      // possible line: 0000000000000000: 00000002 00000000 00010000 0001 01 5841881 @webview_devtools_remote_zeus
+      // the result: match[1] = ''
+      if (!match || !match[1]) {
+        const webView = { pid: -1, pkg: '', socketName };
+        this._webViews.set(socketName, webView);
```

Comment:
It'd be really nice if we could only do it in one place. Extracting a method that computes pkg would really help.

Reply:
> As per my last review, lets extract the function that computes pkg to keep webview bookkeeping in one place. Otherwise looks great!

Thank you for code review. I agree with you. I extracted a function as a class method because it use `this.backend`.